owncloudApp/src/main/java/com/owncloud/android/sharees/presentation/SearchShareesFragment.kt

```diff
@@ -0,0 +1,224 @@
+ */
+class SearchShareesFragment : Fragment(), ShareUserListAdapter.ShareUserAdapterListener {
+
+    // Parameters
+    private var file: OCFile? = null
+    private var account: Account? = null
+
+    // other members
+    private var userGroupsAdapter: ShareUserListAdapter? = null
+    private var listener: ShareFragmentListener? = null
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (arguments != null) {
+            file = arguments!!.getParcelable(ARG_FILE)
+            account = arguments!!.getParcelable(ARG_ACCOUNT)
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        val view = inflater.inflate(R.layout.search_users_groups_layout, container, false)
+
+        // Allow or disallow touches with other visible windows
+        view.filterTouchesWhenObscured = PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(context)
+
+        // Get the SearchView and set the searchable configuration
+        val searchView = view.findViewById<SearchView>(R.id.searchView)
+        val searchManager = activity!!.getSystemService(Context.SEARCH_SERVICE) as SearchManager
+        searchView.setSearchableInfo(
+            searchManager.getSearchableInfo(
+                activity!!.componentName
+            )   // assumes parent activity is the searchable activity
+        )
+        searchView.setIconifiedByDefault(false)    // do not iconify the widget; expand it by default
+
+        searchView.imeOptions = EditorInfo.IME_FLAG_NO_EXTRACT_UI // avoid fullscreen with softkeyboard
+
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String): Boolean {
+                Log_OC.v(TAG, "onQueryTextSubmit intercepted, query: $query")
+                return true    // return true to prevent the query is processed to be queried;
+                // a user / group will be picked only if selected in the list of suggestions
+            }
+
+            override fun onQueryTextChange(newText: String): Boolean {
+                return false   // let it for the parent listener in the hierarchy / default behaviour
+            }
+        })
+
+        return view
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        activity!!.setTitle(R.string.share_with_title)
+
+        // Load private shares in the list
+        listener?.refreshPrivateShares()
+    }
+
+    fun updatePrivateShares(privateShares: ArrayList<OCShare>) {
+        // Update list of users/groups
+        userGroupsAdapter = ShareUserListAdapter(
+            activity!!.applicationContext,
```

Comment:
instead of `activity!!` use `requireActivity()`

Reply:
Done